rosidl_typesupport_opensplice_cpp/CMakeLists.txt

```diff
@@ -23,6 +27,17 @@ ament_export_include_directories(include include/${PROJECT_NAME}/impl)
 
 ament_python_install_package(${PROJECT_NAME})
 
+add_library(${PROJECT_NAME} SHARED src/identifier.cpp)
```

Comment:
I guess that's fine. This solution doesn't leave me feeling particularly good about the overhead and complexity for such a simple check, but I don't see a better way to do it with using a manually maintained implementation id in a header.

> Moving the definition of the symbol to the rmw package would imply that a message package builds a library which contains a declared symbol which is not defined in any of its dependencies.

This is how the `rmw` implementations work, excluding the defaults. `rmw` provides declarations and `rmw_<implementation>` provides the definitions, and the client library package will depend on `rmw` but does not necessarily need to depend on `rmw_<implementation>`. Plus the client libraries use symbols who's implementation exists in the generated message libraries which it does not directly depend on; they're only available when a user depends on both the message library and the client library at the same time. So I don't think it is that odd. But I see the point.

> And we don't want to introduce a dependency in that direction.

I'm not saying we should do it, but you should really back up assertions like this. Though it may be clear to you, it isn't clear to me why this is true.